js/components/addEditBookmark.js

```diff
@@ -23,7 +23,17 @@ class AddEditBookmark extends ImmutableComponent {
     this.onClick = this.onClick.bind(this)
     this.onSave = this.onSave.bind(this)
     this.onRemoveBookmark = this.onRemoveBookmark.bind(this)
+    this.state = {
+      bookmarkNameValid: true,
+      saveButtonDisabled: true
+    };
```

Comment:
I'd recommend that you have an accessor defined for this instead:

```
get bookmarkNameValid () {
...
}
```

Note that there is an onChange handler on the bookmark name so the site data is always known, so you can just check it there in `this.props.currentDetail`.

I think you don't need to know whether the bookmark name is valid separately from if the save button should be disabled. In particular the save button should be disabled when the bookmark name is invalid.